dnf5/context.cpp

```diff
@@ -368,10 +368,47 @@ std::vector<libdnf::rpm::Package> Context::add_cmdline_packages(const std::vecto
 
     if (!packages_paths.empty()) {
         auto cmdline_repo = base.get_repo_sack()->get_cmdline_repo();
+
+        std::vector<std::string> urls;
+
         for (const auto & path : packages_paths) {
+            if (is_url(path)) {
+                urls.emplace_back(path);
+                continue;
+            }
+            // Add local rpm files
             added_packages.push_back(cmdline_repo->add_rpm_package(path, true));
         }
 
+        // Download and add remote rpm files
+        if (!urls.empty()) {
+            std::filesystem::path cmd_repo_pkgs_dir{base.get_config().cachedir().get_value()};
+            cmd_repo_pkgs_dir /= "commandline";
+            cmd_repo_pkgs_dir /= "packages";
+
+            // Ensure that the command line repository packages directory exists.
+            std::filesystem::create_directories(cmd_repo_pkgs_dir);
+
+            // Create a mapping of URLs to paths to destination local files.
+            std::vector<std::pair<std::string, std::filesystem::path>> url2dest_path;
+            for (const auto & url : urls) {
+                std::filesystem::path path(url.substr(url.find("://") + 3));
```

Comment:
I don't think you can just do this. There's a lot more to an URL than just the path, the URL should be properly parsed... There's also an issue with potentially two urls having the package name (weird case, but possible?) and I've also seen URLs which didn't have the filename in the path but still returned an rpm file in the HTTP response.

Reply:
Yes. There exists not handled corner cases. And I think the old DNF also does not handle them correctly. Yes, we want to be better then the old DNF. I have to think about it.

Reply:
I will add TODO